addons/sourcemod/scripting/nd_structure_intercept.sp

```diff
@@ -0,0 +1,212 @@
+    Call_Finish(aBuildStructure);
+
+    switch (aBuildStructure)
+    {
+        case Plugin_Stop:
+        {
+            UTIL_SendBuildingFailed(iClient, iID);
+            return MRES_Supercede;
+        }
+        case Plugin_Changed:
+        {
+            DHookSetParam(hParams, BUILD_PARAM_STRUCTURE, eStructure);
+            DHookSetParamVector(hParams, BUILD_PARAM_POSITION, fOrigin);
+            DHookSetParamVector(hParams, BUILD_PARAM_ANGLES, fAngles);
+            return MRES_ChangedHandled;
+        }
+        case Plugin_Handled:
+        {
+            return MRES_Handled;
+        }
+    }
+
+    // aBuildStructure = Plugin_Continue
+    return MRES_Ignored;
+}
+
+MRESReturn Detour_PlayerBuildEmergencyAssembler(int iClient, DHookParam hParams)
+{
+    eNDStructures eStructure = eND_Assembler;
+    float fOrigin[3];
+    DHookGetParamVector(hParams, BUILD_EMERGENCY_ASSEMBLER_PARAM_POSITION, fOrigin);
+    float fAngles[3];
+    DHookGetParamVector(hParams, BUILD_EMERGENCY_ASSEMBLER_PARAM_ANGLES, fAngles);
+    int iID = DHookGetParam(hParams, BUILD_EMERGENCY_ASSEMBLER_PARAM_ID);
+
+    // receive feedback from other plugins
+    Action aBuildStructure;
+    Call_StartForward(gH_Frwd_OnCommanderBuildStructure);
+    Call_PushCell(iClient);
+    Call_PushCellRef(eStructure);
+    Call_PushArrayEx(fOrigin, sizeof(fOrigin), SM_PARAM_COPYBACK);
+    Call_Finish(aBuildStructure);
+
+    switch (aBuildStructure)
+    {
+        case Plugin_Stop:
+        {
+            UTIL_SendBuildingFailed(iClient, iID);
+            return MRES_Supercede;
+        }
+        case Plugin_Changed:
+        {
+            if (eStructure != eND_Assembler)
+            {
+                LogError("Cannot change emergency assembler build structure to non-assembler structure type.");
+                return MRES_Ignored;
+            }
+            DHookSetParamVector(hParams, BUILD_EMERGENCY_ASSEMBLER_PARAM_ANGLES, fAngles);
+            DHookSetParamVector(hParams, BUILD_EMERGENCY_ASSEMBLER_PARAM_POSITION, fOrigin);
+            return MRES_ChangedHandled;
+        }
+        case Plugin_Handled:
+        {
+            return MRES_Handled;
+        }
+    }
+
+    // aBuildStructure = Plugin_Continue
+    return MRES_Ignored;
+}
+
+#if !defined _nd_commmander_build_included_
+    // This helper function will display red text and a failed sound to the commander
+    stock void UTIL_Commander_FailureText(int iClient, char sMessage[64])
+    {
+        ClientCommand(iClient, "play buttons/button7");
+
+        Handle hBfCommanderText;
+        hBfCommanderText = StartMessageOne("CommanderNotice", iClient, USERMSG_BLOCKHOOKS);
+        BfWriteString(hBfCommanderText, sMessage);
+        EndMessage();
+
+        // clear other messages from notice area
+        hBfCommanderText = StartMessageOne("CommanderNotice", iClient, USERMSG_BLOCKHOOKS);
+        BfWriteString(hBfCommanderText, "");
+        EndMessage();
+    }
+#endif
```

Comment:
This entire block of code is redundant. It's a stock which is never used inside the plugin. It's not going to create.

Reply:
Addressed in https://github.com/stickz/Redstone/commit/4a806ac48ffbdbd305ccc1895f9e8f4925e23b15